consensus/src/simplex/actors/voter/actor.rs

```diff
@@ -923,10 +959,36 @@ impl<
         let payload = proposal.payload;
         let round = self.views.get_mut(&context.view()).unwrap();
         round.requested_proposal_verify = true;
-        Some((
-            context.clone(),
-            self.automaton.verify(context, payload).await,
-        ))
+        let receiver = self.automaton.verify(context.clone(), payload).await;
+        Some(Request { context, receiver })
+    }
+
+    async fn certify(&mut self) -> Option<Request<D, bool>> {
+        let round = self.views.get_mut(&self.view)?;
+
+        // Request certification only once.
+        if round.requested_proposal_certify {
+            return None;
+        }
+        round.requested_proposal_certify = true;
```

Comment:
I think we'd want to set this after actually sending the certify request? Otherwise, a missing proposal will deadlock things I think.